src/Acuminator/Acuminator.Analyzers/StaticAnalysis/PXOverrideMismatch/PXOverrideMismatchAnalyzer.cs

```diff
@@ -0,0 +1,167 @@
+﻿#nullable enable
+
+using System.Collections.Immutable;
+using Acuminator.Analyzers.StaticAnalysis.PXGraph;
+using Acuminator.Utilities.Roslyn.Semantic;
+using Microsoft.CodeAnalysis;
+using Microsoft.CodeAnalysis.Diagnostics;
+using System.Linq;
+using Acuminator.Utilities.DiagnosticSuppression;
+using Acuminator.Utilities.Roslyn.Semantic.PXGraph;
+using Acuminator.Utilities.Common;
+using System.Collections.Generic;
+
+namespace Acuminator.Analyzers.StaticAnalysis.PXOverrideMismatch
+{
+	[DiagnosticAnalyzer(LanguageNames.CSharp)]
+	public class PXOverrideMismatchAnalyzer : PXGraphAggregatedAnalyzerBase
+	{
+		public override ImmutableArray<DiagnosticDescriptor> SupportedDiagnostics =>
+			ImmutableArray.Create(Descriptors.PX1096_PXOverrideMustMatchSignature);
+
+		public override bool ShouldAnalyze(PXContext pxContext, PXGraphSemanticModel graph) =>
+			base.ShouldAnalyze(pxContext, graph) && graph.Type == GraphType.PXGraphExtension;
+
+		public override void Analyze(SymbolAnalysisContext context, PXContext pxContext, PXGraphSemanticModel pxGraphExtension)
+		{
+			context.CancellationToken.ThrowIfCancellationRequested();
+
+			var allMembers = pxGraphExtension.Symbol.GetMembers();
+
+			if (!allMembers.IsDefaultOrEmpty)
+			{
+				var methodsWithPxOverrideAttribute = allMembers
+					.OfType<IMethodSymbol>()
+					.Where(m => m.HasAttribute(pxContext.AttributeTypes.PXOverrideAttribute, checkOverrides: false))
+					.ToList();
+
+				if (methodsWithPxOverrideAttribute.Any())
+				{
+					var allBaseTypes = pxGraphExtension.Symbol
```

Comment:
After discussion with colleagues about the PXOverride behavior in the system it seems that we should add a bit extra logic to exclude direct base types of the extension from the list of all base types in which we search for the PXOverride. 

Imho it should be something like this:
```csharp
if (methodsWithPxOverrideAttribute.Any())
{
      var directBaseTypesAndThis = pxGraphExtension.Symbol.GetBaseTypesAndThis().ToList(capacity: 4);
      var allBaseTypes = pxGraphExtension.Symbol
                                   .GetGraphExtensionWithBaseExtensions(pxContext, SortDirection.Ascending, includeGraph: true)
				  .OfType<INamedTypeSymbol>()
				  .Where(baseType => !directBaseTypesAndThis .Contains(baseType));

     //...
}
```

The rest of PR now looks good to me

Reply:
Changes are pushed.